src/common/vpc-utils.ts

```diff
@@ -39,4 +39,21 @@ export function getExistVpc(scope: Construct, id: string, props: VpcProps): IVpc
   );
 
   return vpc;
+}
+
+export function getALBSubnets(publicSubnetIds: string[], privateSubnetIds: string[]) {
```

Comment:
Because it's token, we cannot compare them in build time(`cdk synth`). If we do know it's actual value(true/false), we have to build different stacks for different inputs.  We have to use the CFN condition to implement it.

Reply:
For us, we require at least two subnets for public and private subnets. We could compare the equal of the first and second items of public and private subnets. It's not perfect, but it should work.